### PR TITLE
LIMS-6296: create 2 suborders with two testplans of same name

### PR DESCRIPTION
**Pull request by ghada-37**
Orders: Create Approach: Make sure In case you create two test plans with the same name and different materiel type, the test units that belongs to them displayed correct in analysis step two
https://modeso.atlassian.net/browse/LIMS-6296

```
➜  1lims-automation git:(LIMS-6296) ✗ nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test006_orders.py:OrdersTestCases.test101_create_order_with_test_plans_with_same_name
2020-09-22 15:12:25.611 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-22 15:12:25.611 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-22 15:12:26.225 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : 8lKXwcAJGEHzFdmbRDaJFdnohzh0tIlQxwctUQ1LKW4 .....
Orders: Create Approach: Make sure In case you create two test plans with the same name ...     
2020-09-22 15:12:46.713 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test101_create_order_with_test_plans_with_same_name
2020-09-22 15:12:50.486 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:603 - wait until page is loaded
2020-09-22 15:12:51.321 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 15:12:52.058 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 15:12:52.775 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:437 - set test unit configuration
2020-09-22 15:12:53.282 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-22 15:12:53.282 | INFO     | api_testing.apis.orders_api:set_configuration:498 - set order configuration
2020-09-22 15:12:53.743 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-22 15:12:53.744 | INFO     | api_testing.apis.test_plan_api:create_double_completed_testplan_same_name_diff_material:361 - create completed test plan with random data
2020-09-22 15:12:53.744 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/articles
2020-09-22 15:12:54.275 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:54.276 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 15:12:54.373 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:54.374 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 15:12:54.528 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:54.529 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/147
2020-09-22 15:12:54.633 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:54.634 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans
2020-09-22 15:12:54.797 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:54.797 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 15:12:54.897 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:54.897 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 15:12:55.002 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:55.003 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 15:12:55.141 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:55.142 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/148
2020-09-22 15:12:55.243 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:55.244 | INFO     | api_testing.apis.test_plan_api:create_double_completed_testplan_same_name_diff_material:372 - create completed test plan with same name 41225d8caa and new material Process Control
2020-09-22 15:12:55.244 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/articles
2020-09-22 15:12:55.489 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:55.489 | INFO     | api_testing.apis.article_api:get_formatted_article_with_formatted_material_type:207 - search for article with material type {'id': 4, 'text': 'Process Control'}
2020-09-22 15:12:55.490 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans
2020-09-22 15:12:55.636 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 400
2020-09-22 15:12:55.636 | ERROR    | api_testing.apis.base_api:wrapper:139 - {'message': 'name_already_exist', 'status': 400, 'testPlanDetails': {'id': 41651, 'testPlanId': 41651, 'attachments': '[]'}}
2020-09-22 15:12:55.637 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test101_create_order_with_test_plans_with_same_name:3575 - create order with two suborders
2020-09-22 15:12:55.637 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/orders/auto/generatedOrderNumber/?yearOption=1
2020-09-22 15:12:55.739 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:55.740 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/contacts
2020-09-22 15:12:55.856 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:55.857 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/articles
2020-09-22 15:12:56.227 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:56.228 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/materialTypes
2020-09-22 15:12:56.331 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:56.332 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits
2020-09-22 15:12:56.478 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:56.479 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testUnits/get/149
2020-09-22 15:12:56.579 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:56.580 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/testPlans
2020-09-22 15:12:56.724 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:56.725 | INFO     | api_testing.apis.base_api:wrapper:130 - GET : https://automation.1lims.com/api/orders
2020-09-22 15:12:56.892 | DEBUG    | api_testing.apis.base_api:wrapper:137 - status code: 1
2020-09-22 15:12:56.892 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test101_create_order_with_test_plans_with_same_name:3579 - open edit page of order 66
2020-09-22 15:12:58.959 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:603 - wait until page is loaded
2020-09-22 15:12:59.336 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 15:13:00.070 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 15:13:00.794 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test101_create_order_with_test_plans_with_same_name:3581 - Navigate to analysis step 2
2020-09-22 15:13:01.114 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:603 - wait until page is loaded
2020-09-22 15:13:01.587 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-22 15:13:02.337 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test101_create_order_with_test_plans_with_same_name:3584 - assert that only 2 analysis triggered
2020-09-22 15:13:02.587 | DEBUG    | ui_testing.pages.base_pages:sleep_medium:47 - wait up to 2 sec
2020-09-22 15:13:06.643 | DEBUG    | ui_testing.pages.base_pages:sleep_medium:47 - wait up to 2 sec
2020-09-22 15:13:10.358 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-22 15:13:12.496 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 45.740s

OK

```

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 1]
```diff
@@ -3213,5 +3214,57 @@ def test093_select_large_number_of_test_units_in_one_testplan(self):
...
+
+        test_units_list = [testplan1['testUnits'][0]['name'], formated_testunit['name']]
+        response, testplan2 = self.test_plan_api.create_testplan(selectedTestPlan=test_plan_name_dict,
+                                                                 testPlan=test_plan_name_dict,
+                                                                 testUnits=[formated_testunit],
+                                                                 selectedArticles=[formatted_article],
+                                                                 materialType=[formatted_material],
+                                                                 materialTypeId=[new_material_id])
+        self.assertEqual(response['message'], 'name_already_exist')
+        self.order_page.create_new_order(material_type=testplan1['materialType'][0]['text'],
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test006_orders.py:3251]: create order using API

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 2]
```diff
@@ -3213,5 +3214,57 @@ def test093_select_large_number_of_test_units_in_one_testplan(self):
             if result['test_plan'] == testPlan['testPlan']['text']:
                 for testunit in testunit_names:
                     self.assertIn(testunit, result['test_units'])
+    
+    @parameterized.expand(['before', 'after'])
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test006_orders.py:3218]: no need to check before/after the saving

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 3]
```diff
@@ -3213,5 +3214,57 @@ def test093_select_large_number_of_test_units_in_one_testplan(self):
...
+                                                                 materialType=[formatted_material],
+                                                                 materialTypeId=[new_material_id])
+        self.assertEqual(response['message'], 'name_already_exist')
+        self.order_page.create_new_order(material_type=testplan1['materialType'][0]['text'],
+                                         article=testplan1['selectedArticles'][0]['text'],
+                                         test_plans=[testplan1['testPlan']['text']],
+                                         with_testunits=False, save=False)
+        if save_before == 'before':
+            self.order_page.save(save_btn='order:save_btn')
+        self.order_page.create_new_suborder(material_type=new_material,
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test006_orders.py:3257]: use API

**Review by 0xIslamTaha — CHANGES_REQUESTED** [review]

File: ui_testing/pages/order_page.py [hunk 4]
```diff
@@ -364,7 +366,7 @@ def get_suborder_table(self):
         self.base_selenium.click(element='order:suborder_list')
 
     def create_new_suborder(self, material_type='', article_name='', test_plan='', test_unit='',
-                            add_new_suborder_btn='order:add_new_item'):
+                            add_new_suborder_btn='order:add_new_item', with_test_unit=True):
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/pages/order_page.py:369]: done

File: ui_testing/pages/order_page.py [hunk 5]
```diff
@@ -386,7 +388,8 @@ def create_new_suborder(self, material_type='', article_name='', test_plan='', t
         self.info('Set test plan : {}'.format(test_plan))
         self.set_test_plan(test_plan)
         self.sleep_tiny()
-        self.set_test_unit(test_unit)
+        if with_test_unit:
+            self.set_test_unit(test_unit)
```
[the remark this answers is not in the document]
Comment (ghada-37) [reply ui_testing/pages/order_page.py:392]: done

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 1 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py:3251]: done

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 2 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py:3218]: done

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 3 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py:3257]: done

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 6]
```diff
@@ -3537,6 +3538,58 @@ def test100_create_multiple_suborders_with_testplans_testunits(self):
...
+                            'materialType': first_material,
+                            'materialTypeId': first_material['id'],
+                            'article': first_article,
+                            'articleId': first_article['id']},
+                           {'testPlans': [second_test_plan_dict],
+                            'selectedTestPlans': [second_test_plan_dict],
+                            'materialType': second_material,
+                            'materialTypeId': second_material['id'],
+                            'article': second_article,
+                            'articleId': second_article['id']}]
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test006_orders.py:3573]: Create a method that takes the test_plan_list and returns the suborder_data

**Review by 0xIslamTaha — CHANGES_REQUESTED** [review]

File: ui_testing/testcases/basic_tests/test006_orders.py (hunk 6 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py]: done

**Pull request merged**